Gems/RecastNavigation/Code/Source/Misc/RecastNavigationPhysXProviderCommon.cpp

```diff
@@ -0,0 +1,232 @@
+        bool debugDrawInputData)
+    {
+        AZ_PROFILE_SCOPE(Navigation, "Navigation: AppendColliderGeometry");
+
+        AZStd::vector<AZ::Vector3> vertices;
+        AZStd::vector<AZ::u32> indices;
+        AZStd::size_t indicesCount = geometry.m_indices.size();
+
+        AzPhysics::SceneInterface* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+        AzPhysics::SceneHandle sceneHandle = sceneInterface->GetSceneHandle(GetSceneName());
+
+        for (const auto& overlapHit : overlapHits)
+        {
+            AzPhysics::SimulatedBody* body = sceneInterface->GetSimulatedBodyFromHandle(sceneHandle, overlapHit.m_bodyHandle);
+            if (!body)
+            {
+                continue;
+            }
+
+            AZ::Transform t = AZ::Transform::CreateFromQuaternionAndTranslation(body->GetOrientation(), body->GetPosition());
+            overlapHit.m_shape->GetGeometry(vertices, indices, nullptr);
+            // Note: geometry data is in local space
+
+            if (!vertices.empty())
+            {
+                if (indices.empty())
+                {
+                    // Some PhysX colliders (convex shapes) return geometry without indices. Build indices now.
+                    AZStd::vector<AZ::Vector3> transformed;
+
+                    int currentLocalIndex = 0;
+                    for (const AZ::Vector3& vertex : vertices)
+                    {
+                        const AZ::Vector3 translated = t.TransformPoint(vertex);
+                        geometry.m_vertices.push_back(RecastVector3(translated));
+
+                        if (cl_navmesh_showInputData || debugDrawInputData)
+                        {
+                            transformed.push_back(translated);
+                        }
+
+                        geometry.m_indices.push_back(aznumeric_cast<AZ::u32>(indicesCount + currentLocalIndex));
+                        currentLocalIndex++;
+                    }
+
+                    if (cl_navmesh_showInputData || debugDrawInputData)
+                    {
+                        for (size_t i = 2; i < vertices.size(); i += 3)
+                        {
+                            DebugDraw::DebugDrawRequestBus::Broadcast(&DebugDraw::DebugDrawRequests::DrawLineLocationToLocation,
+                                transformed[i - 2], transformed[i - 1], AZ::Colors::Red, cl_navmesh_showInputDataSeconds);
+                            DebugDraw::DebugDrawRequestBus::Broadcast(&DebugDraw::DebugDrawRequests::DrawLineLocationToLocation,
+                                transformed[i - 1], transformed[i - 0], AZ::Colors::Red, cl_navmesh_showInputDataSeconds);
+                            DebugDraw::DebugDrawRequestBus::Broadcast(&DebugDraw::DebugDrawRequests::DrawLineLocationToLocation,
+                                transformed[i - 0], transformed[i - 2], AZ::Colors::Red, cl_navmesh_showInputDataSeconds);
+                        }
+                    }
```

Comment:
suggestion: Consider moving this to a function or breaking up other parts of the logic here where it makes sense

Reply:
Good idea, done.